### PR TITLE
docs(repo): adds json formatter docs

### PR DESCRIPTION
**Pull request by mnaumanali94**
**Checklist**

- [ ] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

File: docs/guides/2-cli.md [hunk 1]
```diff
@@ -70,6 +70,93 @@ The default behavior can be modified with the `--fail-severity=` option. Setting
...
+      type: array
+      description: An array of strings that indicate the location within the analyzed document where the rule was triggered. It shows the "path" in the document structure to the issue.
+      items:
+        type: string
+    message:
+      type: string
+      description: A string that contains a human-readable message describing the issue found by Spectical. This message typically provides information on why the rule was triggered and how to fix the issue.
+    severity:
+      type: integer
+      description: 'An integer representing the severity level of the rule violation. The severity levels usually follow a specific scale defined by Spectral, such as 0 for info, 1 for warn, 2 for error, etc.'
```
Comment (P0lip) [inline docs/guides/2-cli.md]: 0 for error, 1 for warn, 2 for info, 3 for hint

File: docs/guides/2-cli.md [hunk 2]
```diff
@@ -70,6 +70,93 @@ The default behavior can be modified with the `--fail-severity=` option. Setting
...
+<!--
+type: tab
+title: Schema
+-->
+
+```yaml json_schema
+type: array
+items:
+  type: object
+  properties:
```
Comment (P0lip) [inline docs/guides/2-cli.md:90]: and make everything required mister

File: docs/guides/2-cli.md [hunk 3]
```diff
@@ -70,6 +70,93 @@ The default behavior can be modified with the `--fail-severity=` option. Setting
...
+      description: An array of strings that indicate the location within the analyzed document where the rule was triggered. It shows the "path" in the document structure to the issue.
+      items:
+        type: string
+    message:
+      type: string
+      description: A string that contains a human-readable message describing the issue found by Spectical. This message typically provides information on why the rule was triggered and how to fix the issue.
+    severity:
+      type: integer
+      description: 'An integer representing the severity level of the rule violation. The severity levels usually follow a specific scale defined by Spectral, such as 0 for info, 1 for warn, 2 for error, etc.'
+    range:
```
Comment (P0lip) [inline docs/guides/2-cli.md:109]: and maybe mention it's 0 indexed

File: docs/guides/2-cli.md [hunk 4]
```diff
@@ -70,6 +70,93 @@ The default behavior can be modified with the `--fail-severity=` option. Setting
 
 Changing the fail severity wont' affect output. To change the results Spectral CLI prints to the screen, add the `--display-only-failures` switch (or just `-D` for short). This removes any results which are below the specified fail severity.
 
+## Formatters
+
+### JSON Formatter
+
+Spectral's JSON formatter outputs the results of a Spectral analysis in a JSON format that is easily parsable and human-readable. The output can be used to programmatically access and process the results of the analysis. You can enable this by adding `-f json` to the cli command.
```
Comment (P0lip) [inline docs/guides/2-cli.md]:
```suggestion
Spectral's JSON formatter outputs the results of a Spectral analysis in a JSON format that is easily parsable and human-readable. The output can be used to programmatically access and process the results of the analysis. You can enable this by adding `-f json --quiet` to the cli command.
```

**Review by P0lip — DISMISSED** [review]

**Review by ryotrellim — DISMISSED** [review]
Love this!

**Review by P0lip — APPROVED** [review]

**Pull request merged**